backend/internal/db/migrations/003_add_indexes.sql

```diff
@@ -0,0 +1,65 @@
+-- Migration: Add performance indexes
+-- Description: Create indexes to optimize common queries
+
+-- Nodes table indexes
+-- Index for node type filtering
+CREATE INDEX IF NOT EXISTS idx_nodes_node_type ON nodes(node_type);
+
+-- Index for tags (using GIN for array operations)
+CREATE INDEX IF NOT EXISTS idx_nodes_tags ON nodes USING GIN(tags);
+
+-- Index for file path lookups
+CREATE INDEX IF NOT EXISTS idx_nodes_file_path ON nodes(file_path);
+
+-- Full-text search index on title and content
+CREATE INDEX IF NOT EXISTS idx_nodes_search ON nodes 
+USING GIN(to_tsvector('english', COALESCE(title, '') || ' ' || COALESCE(content, '')));
+
+-- Index for centrality-based sorting (for finding important nodes)
+CREATE INDEX IF NOT EXISTS idx_nodes_centrality ON nodes(centrality DESC);
+
+-- Composite index for common queries
+CREATE INDEX IF NOT EXISTS idx_nodes_type_centrality ON nodes(node_type, centrality DESC);
+
+-- Edges table indexes
+-- Index for source node lookups
+CREATE INDEX IF NOT EXISTS idx_edges_source_id ON edges(source_id);
+
+-- Index for target node lookups
+CREATE INDEX IF NOT EXISTS idx_edges_target_id ON edges(target_id);
+
+-- Composite index for source-target lookups
+CREATE INDEX IF NOT EXISTS idx_edges_source_target ON edges(source_id, target_id);
```

Comment:
I noticed there's a duplicate index here. Line 32 creates `idx_edges_source_target` on `edges(source_id, target_id)`, but line 24 in the previous migration (002_vault_models.sql) already creates the same index with the same name. This will likely cause an error unless IF NOT EXISTS is handling it gracefully.